olp-cpp-sdk-core/src/http/curl/NetworkCurl.cpp

```diff
@@ -579,21 +595,22 @@ ErrorCode NetworkCurl::SendImplementation(
 }
 
 void NetworkCurl::Cancel(RequestId id) {
-  OLP_SDK_LOG_TRACE(kLogTag, "cancel with id = " << id);
-
   if (!IsStarted()) {
+    OLP_SDK_LOG_ERROR(kLogTag, "Can't cancel the request id="
+                                   << id << " - network is not started");
     return;
   }
   std::lock_guard<std::mutex> lock(event_mutex_);
   for (auto& handle : handles_) {
     if (handle.in_use && (handle.id == id)) {
       handle.cancelled = true;
       AddEvent(EventInfo::Type::CANCEL_EVENT, &handle);
+
+      OLP_SDK_LOG_TRACE(kLogTag, "cancel request id=" << id);
       return;
     }
   }
-  OLP_SDK_LOG_WARNING(kLogTag, "cancel for non-existing request " << id);
-  return;
+  OLP_SDK_LOG_WARNING(kLogTag, "cancel for non-existing request id=" << id);
```

Comment:
Maybe this should not be an warning but rather an debug.
Nothing to cancel is not an error but rather: job done.

Reply:
Maybe we should. But because `Cancel` doesn't provide any instant feedback for the user (like returning bool) I think, we should 'warn' the user that they is doing something wrong.